Plugins/ExternalTexture/Source/ExternalTexture_Shared.h

```diff
@@ -1,7 +1,34 @@
-// Shared pimpl code (not an actual header)
+// Shared code (not an actual header)
 
 namespace Babylon::Plugins
 {
+    ExternalTexture::Impl::Impl(Graphics::TextureT ptr)
+    {
+        Init(ptr);
+        GetInfo(m_width, m_height, m_hasMips, m_format, m_flags);
+    }
+
+    void ExternalTexture::Impl::Update(Graphics::TextureT ptr)
+    {
+        Init(ptr);
+
+#ifndef NDEBUG
+        uint16_t width{};
+        uint16_t height{};
+        bgfx::TextureFormat::Enum format{bgfx::TextureFormat::Unknown};
+        uint64_t flags{};
+        bool hasMips{};
+        GetInfo(width, height, hasMips, format, flags);
+        assert(width == m_width);
+        assert(height == m_height);
+        assert(format == m_format);
+        assert(flags == m_flags);
+        assert(hasMips == m_hasMips);
```

Comment:
Should these really only be debug asserts and not runtime errors?

Reply:
I could make it a runtime error, but I'm not sure what impact it will have on perf.